### PR TITLE
Add information about Trilinos stacks

### PR DESCRIPTION
**Pull request by jpthiele**
This is the start of adding actual content to the Trilinos section.

I possibly went into a bit more detail than needed but I always find it easier to condense that
than to add after the fact.

### CODE REVIEW

**Comment by jpthiele** [comment]
@kinnewig FYI

File: 9.6/paper.tex [hunk 1]
```diff
@@ -312,9 +318,33 @@ \subsection{Advances in non-matching support}\label{sec:nonmatching}
...
+and the newer \tpetra{} (templated Petra) providing additional shared memory 
+parallelism and GPU capabilities through \kokkos{}, referred to as \MPIx.
+Since the interface of \tpetra{} was overhauled and modernized 
+many of the packages in the third layer now have two generations as well.
+Therefore, the \MPI-only set of packages is also referred to as the 
+\epetra{} stack and the \MPIx{} packages form the \tpetra{} stack.
+As a consequence of the different interfaces, 
+a decision was made to deprecate the \epetra{} stack 
+and only provide basic maintenance until a full 
+removal from the repository in the course of 2025.
```
Comment (masterleinad) [inline 9.6/paper.tex:345]: We have to make clear that we are talking about `Trilinos`, not `deal.II` here.

Comment (jpthiele) [reply 9.6/paper.tex:345]:
```suggestion
As a consequence of the different interfaces, 
a decision was made in the \trilinos project to deprecate the \epetra{} stack 
and only provide basic maintenance until a full 
removal from the \trilinos repository in the course of 2025.
```

Comment (jpthiele) [reply 9.6/paper.tex]: Agreed, better?

Comment (masterleinad) [reply 9.6/paper.tex]: Yeah

**Comment by bangerth** [comment]
Let's merge. We can always all take turns improving the text.

**Pull request merged**